base/strings/basic.jl

```diff
@@ -610,3 +610,52 @@ function filter(f, s::AbstractString)
     end
     String(take!(out))
 end
+
+## string first and last ##
+
+"""
+    first(str::AbstractString, nchar::Integer)
+
+Get a string consisting of the first `nchar` characters of `str`.
+
+```jldoctest
+julia> first("∀ϵ≠0: ϵ²>0", 0)
+""
+
+julia> first("∀ϵ≠0: ϵ²>0", 1)
+"∀"
+
+julia> first("∀ϵ≠0: ϵ²>0", 3)
+"∀ϵ≠"
+```
+"""
+function first(str::AbstractString, nchar::Integer)
+    if 0 <= nchar <= 1
+        return str[1:nchar]
+    end
+    str[1:nextind(str, 1, nchar-1)]
```

Comment:
This seems like it will give a `BoundsError` for `nchar > length(str)`.  Shouldn't it be `min(endof(str), nextind(str, 1, nchar-1))`?

Reply:
Or is `BoundsError` intended here?

Reply:
It was intended. The idea is to have an invariant `length(first(s, nchar)) == nchar`. Similarly for `last`.
This might be changed the way you propose - then also docstring should be changed as it now promises `nchar` characters in the produced string.
I slightly prefer the current implementation (as it ensures the invariant) but I can see the rationale behind your proposal so I am OK change it.

Reply:
If we changed it to "at most" nchar elements of the string, then it would be similar to `take`.

My feeling is that using "at most" nchar elements would be a big more flexible, and it wouldn't hurt the uses where you want exactly nchar elements except that it wouldn't throw an error for a string of the wrong size.

Reply:
OK. I will make a PR to have a clear decision point.